### PR TITLE
Build channel removal

### PR DESCRIPTION
**Pull request by mmitche**
Adds the ability to remove builds from channels. There are times when builds get accidentally added to a channel. In these cases, we'd like to NOT do another build, and simply roll back. This is fairly straightforward. The build+channel combo is removed from the BuildChannels channels table, and if it happened to be that the build was the last applied build, then re-triggering the subscription will roll back to previous build.

Added scenario tests that check for this behavior.

### CODE REVIEW

File: src/Maestro/Maestro.Web/Api/v2018_07_16/Controllers/ChannelsController.cs [hunk 1]
```diff
@@ -16,6 +16,7 @@
 using Microsoft.AspNetCore.ApiVersioning.Swashbuckle;
 using Build = Maestro.Data.Models.Build;
 using Channel = Maestro.Web.Api.v2018_07_16.Models.Channel;
+using Castle.Core.Resource;
```
Comment (riarenas) [inline src/Maestro/Maestro.Web/Api/v2018_07_16/Controllers/ChannelsController.cs]: What is this one for?

File: src/Maestro/tests/Scenarios/azdoflow-nonbatched.ps1 [hunk 2]
```diff
@@ -25,6 +26,53 @@ $sourceAssets = @(
     }
 )
 
+$sourceBuild2Number = Get-Random
```
Comment (riarenas) [inline src/Maestro/tests/Scenarios/azdoflow-nonbatched.ps1]: [nit] $source2BuildNumber

File: src/Maestro/tests/Scenarios/azdoflow-nonbatched.ps1 [hunk 3]
```diff
@@ -25,6 +26,53 @@ $sourceAssets = @(
...
+    },
+    @{
+        name = "Bar"
+        version = "2.17.0"
+    }
+)
+
+$expectedDependencies1 =@(
+        "Name:             Foo"
+        "Version:          1.1.0",
```
Comment (riarenas) [inline src/Maestro/tests/Scenarios/azdoflow-nonbatched.ps1]: >1.1.0 [](start = 27, length = 5)

Should we add the slashes to these versions for consistency?

File: src/Maestro/tests/Scenarios/githubflow-nonbatched.ps1 [hunk 4]
```diff
@@ -27,6 +27,53 @@ $sourceAssets = @(
     }
 )
 
+$sourceBuild2Number = Get-Random
```
Comment (riarenas) [inline src/Maestro/tests/Scenarios/githubflow-nonbatched.ps1]: [nit] $source2BuildNumber

File: src/Maestro/Maestro.Web/Api/v2018_07_16/Controllers/ChannelsController.cs (hunk 1 above)
Comment (mmitche) [reply src/Maestro/Maestro.Web/Api/v2018_07_16/Controllers/ChannelsController.cs]: VS keeps doing this...i have no idea why..it just randomly introduces new usings.

**Pull request merged**